src/nupic/algorithms/SpatialPooler.cpp

```diff
@@ -196,6 +196,7 @@ Real SpatialPooler::getPotentialPct() const
 
 void SpatialPooler::setPotentialPct(Real potentialPct)
 {
+  NTA_ASSERT(potentialPct > 0 && potentialPct <= 1); //bounds check
```

Comment:
turn assert to always check? 
We should check the public API and if it's public, always check. Other hand, performance impact.